vtm/src/org/oscim/tiling/source/mapfile/MapDatabase.java

```diff
@@ -406,9 +409,9 @@ private void processBlock(QueryParameters queryParameters,
     //    private long mCurrentRow;
     //    private long mCurrentCol;
 
-    private int xmin, ymin, xmax, ymax;
+    private int xmin, ymin, xmax, ymax, xSmin, ySmin, xSmax, ySmax;
 
-    private void setTileClipping(QueryParameters queryParameters, long mCurrentRow, long mCurrentCol) {
+    private synchronized void setTileClipping(QueryParameters queryParameters, long mCurrentRow, long mCurrentCol) {
```

Comment:
To be sure: I inserted `synchronized` to ensure that private vars `xmin` `ymin` etc. aren't overwritten while they're used. Don't know if it really does what I meant, or if it was missing at all (?)

Reply:
These variables can be converted to local variables inside the method and avoid the `synchronized`?

Reply:
These variables can be converted to local variables inside the method and avoid the `synchronized`?

Reply:
I think so. 
It works without synchronized, too. But don't know if it can lead to inconsistencies.

Reply:
I pushed the changes.